package/MDAnalysis/lib/distances.py

```diff
@@ -208,6 +208,23 @@ def _check_lengths_match(*arrays):
         raise ValueError("Input arrays must all be same shape"
                          "Got {0}".format([a.shape for a in arrays]))
 
+
+def distance(a, b, box=None):
+    """calc distances between two points. Uses algorithm from Tuckerman
+
+    """
+    if box is None:
+        return np.linalg.norm(a - b)
+
+    box = triclinic_vectors(box)
+    ibox = np.linalg.inv(box)
```

Comment:
You can explicitly calculate the inverse as in #1475. 

This is marginally faster in numpy but might be a lot faster in cython.

```python
import numpy as np

def make_box():
    a = [np.random.random(), 0, 0]
    b = np.zeros(3)
    b[:2] = np.random.random(2)
    c = np.random.random(3)
    return 20 * np.array([a, b, c])

def ibox(box):
    a,b,c = box
    return np.array([[1/a[0], 0, 0], [-b[0]/(a[0]*b[1]), 1/b[1], 0], [(b[0]*c[1] - b[1]*c[0])/(a[0]*b[1]*c[2]), -c[1]/(b[1]*c[2]), 1/c[2]]])
```

Timing:
```python
box = make_box()
%timeit ibox(box)
# 7.33 µs ± 108 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
%timeit np.linalg.inv(box)
#9.98 µs ± 247 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Correctness
```python
In [42]: np.dot(box, ibox(box))
Out[42]:
array([[  1.00000000e+00,   0.00000000e+00,   0.00000000e+00],
       [  1.90479818e-18,   1.00000000e+00,   0.00000000e+00],
       [  3.40613054e-17,  -7.40127872e-17,   1.00000000e+00]])
```



(Closed form from sympy...).
```python
from sympy.interactive.printing import init_printing
init_printing(use_unicode=False, wrap_line=False, no_global=True)
a0, b0, b1, c0, c1, c2 = sym.symbols("a0, b0, b1, c0, c1, c2")
h = sym.Matrix([[a0, b0, c0], [0, b1, c1], [0, 0, c2]])
h.inv()

In [35]: h
Out[35]:
[a0  b0  c0]
[          ]
[0   b1  c1]
[          ]
[0   0   c2]

In [37]: h.inv()
Out[37]:
[1    -b0   b0*c1 - b1*c0]
[--  -----  -------------]
[a0  a0*b1     a0*b1*c2  ]
[                        ]
[     1          -c1     ]
[0    --        -----    ]
[     b1        b1*c2    ]
[                        ]
[                1       ]
[0     0         --      ]
[                c2      ]
```